feat(protocol-designer): add DOM identifiers for e2e tests

shlokamin opened the pull request:
# Overview

This PR adds a bunch of identifiers to be used by e2e tests.

closes #7607

# Changelog
- Add DOM identifiers for e2e tests

# Review requests
Code review, make sure names and all that make sense
Make sure identifiers show up on the DOM

# Risk assessment
Low

shlokamin committed:
feat(protocol-designer): add DOM identifiers for e2e tests

#7607

shlokamin commented on protocol-designer/src/components/StepEditForm/fields/PathField/Path.js:
```diff
@@ -89,13 +99,17 @@ const getSubtitle = (
   return reasonForDisabled || ''
 }
 
+const getPathButtonId = (name, selectedValue) =>
+  `PathButton_${name}_${name === selectedValue ? 'selected' : 'deselected'}`
+
```
don't love having whether the path option is selected or deselected in the id, but not sure of a better way to do this

shlokamin commented on protocol-designer/src/components/StepEditForm/fields/WellOrderField/index.js:
```diff
@@ -109,9 +109,14 @@ export const WellOrderField = (props: Props): React.Node => {
                 getIconClassNames()
               )}
+              id="WellOrderField_button"
             />
           ) : (
-            <Text onClick={handleOpen} css={mixedWellOrderStyles}>
+            <Text
+              onClick={handleOpen}
+              css={mixedWellOrderStyles}
+              id="WellOrderField_button"
+            >
```
these two won't ever show up on the DOM at the same time, so its cool to use the same id i think. alternatively, i could do 

`id="WellOrderField_button"` and `id="WellOrderField_button_mixed"` or something

IanLondon reviewed (changes requested):
Missing: within the tip position modal, if you click "custom", the `input type="text"` field where you can enter `mm` still could use a selector.

✔️ Good move: `name=TipPositionOptions` appears on two `input` elements. And that's correct, for radio button `input` elements! I didn't realize radio buttons in a group are supposed to share `name`, I learned something new :D

---

For well order field: I think Neha wanted to be able to "read" the button without opening the modal (which way the zig-zag icon is oriented. In the ticket, we had the idea `data-test="t2b,l2r"` to represent first: top to bottom & second: left to right, for example. to do right to left, top to bottom the `data-test` of the icon/button would be `data-test="r2l,t2b"` and so on)

Also, ids are supposed to be unique but there are 2x `id=WellOrderField_button` elements in Transfer. Common id for aspirate well order vs dispense well order.  Maybe address by doing `WellOrderField_button_${aspirate|dispense}` or something like that?

---

For selector number 6, the path button selectors, I think they're supposed to differentiate disabled vs simply deselected. Right now you can only tell that they're deselected, but I'm pretty sure we want to be able to confirm that they're disabled or not. It might be easier for E2E if the ID is constant for each button (eg `id="PathButton_single"` with no modifiers to the `id`) and `data-test` can be the lookup for selected vs deselected, and disabled or not disabled??

---

For selector number 7, the `disposal volume > blowout` field may need a selector?

---

For selector number 8, would it make more sense for it to be `name="pipette"` rather than using `id="PipetteField_dropdown"` to match eg the `name="aspirate_labware"` and `name="dispense_labware"` dropdowns?

shlokamin committed:
make well order button id unique

shlokamin committed:
PR review comments

shlokamin commented:
Great feedback, thanks @IanLondon. I like this PR much better now 😄 

> Missing: within the tip position modal, if you click "custom", the `input type="text"` field where you can enter `mm` still could use a selector.

Good catch, added `<input type="text" value="" id="TipPositionModal_custom-input">`

> For well order field: I think Neha wanted to be able to "read" the button without opening the modal (which way the zig-zag icon is oriented. In the ticket, we had the idea `data-test="t2b,l2r"` to represent first: top to bottom & second: left to right, for example. to do right to left, top to bottom the `data-test` of the icon/button would be `data-test="r2l,t2b"` and so on)
> 
> Also, ids are supposed to be unique but there are 2x `id=WellOrderField_button` elements in Transfer. Common id for aspirate well order vs dispense well order. Maybe address by doing `WellOrderField_button_${aspirate|dispense}` or something like that?

Gocha, I added data-test but also kept the id so we can more semantically click on the button if need be. Also, to follow our convention of `${ComponentName}_${Indentifer}` where component name is pascal case identifier is snake case I did:


```js
<img id="WellOrderField_button_aspirate" data-test="WellOrderField_button_t2b_l2r">
```


> For selector number 6, the path button selectors, I think they're supposed to differentiate disabled vs simply deselected. Right now you can only tell that they're deselected, but I'm pretty sure we want to be able to confirm that they're disabled or not. It might be easier for E2E if the ID is constant for each button (eg `id="PathButton_single"` with no modifiers to the `id`) and `data-test` can be the lookup for selected vs deselected, and disabled or not disabled??

Oof, this one kinda sucks because we're adding a lot of extra stuff here. I wish we could more semantically query the class name (since thats is actually what is being used to render what the button actually means), but that is not very reliable. Anyway, I did:

```js
<li id="PathButton_single" data-test="PathButton_selected_enabled"></li>
```



> For selector number 7, the `disposal volume > blowout` field may need a selector?

Good call, I added:

```js
<select id="DisposalVolumeField_dropdown">
   <option value="trashId">Trash</option>
   <option value="source_well">Source Well</option>
   <option value="dest_well" disabled="">Destination Well</option>
</select>
```

> For selector number 8, would it make more sense for it to be `name="pipette"` rather than using `id="PipetteField_dropdown"` to match eg the `name="aspirate_labware"` and `name="dispense_labware"` dropdowns?

Another good call! I did:

```js
<select name="pipette">
   <option value="56ac2870-9bae-11eb-8a4b-e7b378696bf6">P20 Single-Channel GEN2</option>
   <option value="56ac2871-9bae-11eb-8a4b-e7b378696bf6">P20 8-Channel GEN2</option>
</select>
```

shlokamin committed:
add id to settings button

IanLondon reviewed (approved):
changes look good!!

Kadee80 reviewed (approved):
🔚 🥈 🔚
